src/app/navigation/middleware.ts

```diff
@@ -31,7 +33,12 @@ export default (routes: AnyRoute[], history: History): Middleware => ({dispatch}
     history[action.payload.method]({
       hash: action.payload.hash,
       pathname: matchPathname(action.payload),
-      search: matchSearch(action.payload, action.payload.search),
+      search: matchSearch(
+        action.payload,
+        {
```

Comment:
so here is the problem malar found, all params are "persistent" so if you try to remove a query param by dispatching a location change without it, it'll be re-added by the middleware.

Reply:
maybe if we're going to follow through with this pattern, we need to put a filter on undefined in here and then the only way to remove a "persistent" query param is to explicitly pass undefined so it does get re-added.

i'm not sure i really like that, but it would work

Reply:
I'm struggling with finding a solution outside of inserting the system params here. Specifically I haven't found a way to persist the `content-style` query param once `locationChange` is dispatched after a search.

Reply:
you'd have to put it back in the location change dispatch in the search hook right?